views/forms/DropdownCheckboxGroup/index.jsx

```diff
@@ -1,89 +1,119 @@
 import React, { useState } from 'react'
-import { useWatch } from 'react-hook-form'
-import { KeyboardArrowDown } from '@mui/icons-material'
-import { Menu, ListItem, TextField, InputAdornment } from '@mui/material'
-import ControlledCheckbox from '../ControlledCheckbox'
+import { FormControl, OutlinedInput, InputLabel, Select, MenuItem, Box, Chip } from '@mui/material'
+
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250,
+    },
+  },
+};
 
 const DropdownCheckboxGroup = ({
-  filterKey,
-  control,
-  options,
-  onSubmit,
-  getValues,
+  label,
+  initialValues,
+  onChange
 }) => {
-  const [anchorEl, setAnchorEl] = useState(null)
-  const open = Boolean(anchorEl)
-  const handleClick = (event) => {
-    setAnchorEl(event.currentTarget)
-  }
-  const handleClose = async () => {
-    setAnchorEl(null)
-
-    const values = getValues()
+  const options = Object.keys(initialValues)
+  const selectedValues = Object.keys(initialValues).filter(k => initialValues[k].value === 1)
+  const [selectedValue, setSelectedValue] = useState(selectedValues)
 
-    await onSubmit(values)
+  const handleChange = (event) => {
+    const {
+      target: { value },
+    } = event;
+    setSelectedValue(value)
+    onChange(label, value)
   }
-
-  const formValues = useWatch({
-    control,
-    name: filterKey,
-  })
-
   return (
-    <>
-      <TextField
-        label={filterKey}
-        id={`${filterKey}`}
-        placeholder="Select"
-        InputLabelProps={{
-          shrink: true,
-          sx: {
-            color: 'grey.A100',
-            '&.Mui-focused': { color: 'grey.A100' },
-            '&:hover': { cursor: 'none' },
-          },
-        }}
-        onClick={handleClick}
-        onFocus={(e) => e.currentTarget.blur()}
-        sx={{
-          borderColor: 'grey.A100',
+    <>      
+      <FormControl sx={{ m: 1, width: 300 }}>
+      <InputLabel id={`multi-chip-label-${label}`}>{label}</InputLabel>
+      <Select
+        labelId={`multi-chip-label-${label}`}
+        id={`multi-chip-${label}`}
+        multiple
+        value={selectedValue}
+        onChange={handleChange}
+        input={<OutlinedInput id={`select-multiple-${label}`} label={label} />}
+        renderValue={(selected) => {
+          const EtcChip = selected.length > 5 ? <Chip key={"etc"} label="..." /> : <></>
```

Comment:
do you need the "key" prop?

Reply:
Yes, the Chip is rendered by a `map` internally